lib/std/unicode.zig

```diff
@@ -62,32 +112,35 @@ pub fn utf8Encode(c: u21, out: []u8) !u3 {
     return length;
 }
 
-const Utf8DecodeError = Utf8Decode2Error || Utf8Decode3Error || Utf8Decode4Error;
-
-/// Decodes the UTF-8 codepoint encoded in the given slice of bytes.
-/// bytes.len must be equal to utf8ByteSequenceLength(bytes[0]) catch unreachable.
-/// If you already know the length at comptime, you can call one of
-/// utf8Decode2,utf8Decode3,utf8Decode4 directly instead of this function.
-pub fn utf8Decode(bytes: []const u8) Utf8DecodeError!u21 {
-    return switch (bytes.len) {
-        1 => @as(u21, bytes[0]),
-        2 => utf8Decode2(bytes),
-        3 => utf8Decode3(bytes),
-        4 => utf8Decode4(bytes),
+/// Decodes the UTF-8 codepoint encoded in the given slice of bytes and returns
+/// then length of the character decoded.
+///
+/// Guaranteed to not read bytes past this character.
+///
+/// "ret" cannot be *u21 because when casting to *u32 it would have differn't
+/// behavior on Little-Endian and Big-Endian machines, which is too much to ask
+/// of our callers.
+pub fn utf8Decode(bytes: []const u8, ret: *align(4) u32) !u3 {
```

Comment:
Why have an outarg here? ==> you could return a struct with `codepoint` and `bytes_read` fields.
Or have bytes_read as the out-arg so we don't have the endian issue.

Reply:
There is no endianness issue.

I do like multiple return values, but zig (and C) does not handle those well in the language. I think we should just support them the way Go does, which is not very intrusive on the language.

Reply:
> There is no endianness issue.

then what is the doc-comment about?